### PR TITLE
DDF-2179 Update configuration file formatting documentation

### PR DESCRIPTION
**Pull request by mweser**
#### What does this PR do?
Updates documentation regarding formatting of string arrays in configuration files.

#### Who is reviewing it (please choose AT LEAST two reviewers that need to approve the PR before it can get merged; if a component team is listed, at least one of its members needs to approve)?
@peterhuffer 
@spearskw 
@ricklarsen 
@lessarderic 

#### Choose 2 committers to review/merge the PR (please choose ONLY two committers from below, delete the rest).
@jaymcnallie
@kcwire
#### Checklist:
- [x] Documentation Updated
- [  ] Update / Add Unit Tests
- [  ] Update / Add Integration Tests

### CODE REVIEW

**Comment by ricklarsen** [comment]
👍

**Comment by peterhuffer** [comment]
LGTM

**Comment by spearskw** [comment]
👍

File: distribution/docs/src/main/resources/_contents/configuring-from-configuration-files-contents.adoc [hunk 1]
```diff
@@ -334,7 +334,7 @@ The following table shows the format to use for each property type supported.
 
 |Lists of Strings
 |`name = [ "value1", "value2", ... ]`
-|`authenticationTypes = [ "/\=SAML\|GUEST", "/admin\=SAML\|basic", "/system\=basic", "/solr\=SAML\|PKI\|basic", "/sources\=SAML\|basic", "/security-config\=SAML\|basic" ]`
+|`complexStringArray = [ "{\"url\"\ \"http://test.sample.com\"\ \"layers\"\ [\"0\"]\ {\"VERSION\"\ \"1.1\|1.2\"\ \"image/png\"}\ \"beta\"\ 1}", "{\"url\"\ \"http://test.sample.com"\ 0.5}", "/solr\=SAML\|PKI\|basic", "/security-config\=SAML\|basic" ]`
```
Comment (stustison) [inline distribution/docs/src/main/resources/_contents/configuring-from-configuration-files-contents.adoc]: This doesn't make any sense. Why did this get changed?

Comment (mweser) [inline distribution/docs/src/main/resources/_contents/configuring-from-configuration-files-contents.adoc]: @stustison This got changed as the example in the documentation does not demonstrate how to handle { } and [ ] characters in formatted string arrays. I had to discover for myself whether they required escape characters as double-quotes and whitespaces did. Below this in the documentation is a note on which characters require escapes.

Comment (stustison) [inline distribution/docs/src/main/resources/_contents/configuring-from-configuration-files-contents.adoc]: Not all of the configurations we have follow the same format. The format you're showing works for the Search UI, but does not work for the Policy Manager.

Comment (lessarderic) [inline distribution/docs/src/main/resources/_contents/configuring-from-configuration-files-contents.adoc]: @stustison: This is just an example of what a configuration can look like and is not specific to the Policy Manager or anything else. We were just trying to show the different things that needed to be escaped or not escaped.

To reduce the confusion, we could replace the `solr`, `SAML`, `PKI`, `basic` and `security-config` with non-security names. Would that work?

**Comment by kcwire** [comment]
👍

File: distribution/docs/src/main/resources/_contents/configuring-from-configuration-files-contents.adoc (hunk 1 above)
Comment (lessarderic) [inline distribution/docs/src/main/resources/_contents/configuring-from-configuration-files-contents.adoc]: @mweser: Let's stick with a simple list example in the table (even simpler than the original) and add more complex examples that show how the characters can be escaped in the notes below the table.

**Comment by lessarderic** [comment]
👍 when the list example in the documentation has been updated.

**Pull request merged**